Neos.Flow/Classes/Http/Helper/ResponseInformationHelper.php

```diff
@@ -149,7 +150,18 @@ public static function prepareHeaders(ResponseInterface $response): array
     public static function makeStandardsCompliant(ResponseInterface $response, RequestInterface $request): ResponseInterface
     {
         $statusCode = $response->getStatusCode();
-        if ($request->hasHeader('If-Modified-Since') && $response->hasHeader('Last-Modified') && $statusCode === 200) {
+        if ($request->hasHeader('If-None-Match') && in_array($request->getMethod(), ['HEAD', 'GET'])
+            && $response->hasHeader('ETag') && $statusCode === 200) {
+            $ifNoneMatchHeaders = $request->getHeader('If-None-Match');
+            $eTagHeader = $response->getHeader('ETag')[0];
+            foreach ($ifNoneMatchHeaders as $ifNoneMatchHeader) {
+                if (ltrim($ifNoneMatchHeader, 'W/') == ltrim($eTagHeader, 'W/')) {
+                    $response = $response->withStatus(304);
+                    break;
+                }
+            }
+        } elseif ($request->hasHeader('If-Modified-Since') && in_array($request->getMethod(), ['HEAD', 'GET'])
```

Comment:
At some point we should also add `If-Unmodified-Since` ;) (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Unmodified-Since)

Reply:
There is a section for that here https://github.com/neos/flow-development-collection/blob/master/Neos.Flow/Classes/Http/Helper/ResponseInformationHelper.php#L160-L169 but that has the same issues that would apply to handling POST requests. 

The If-Match and If-Unmodified-Since headers are specified to prevent two concurrent write operations beeing applied. The status 412 only indicates that the application could not perform the  requested operation because a precondition was not matched. 

Adding only the 412 header as is currently done without preventing the execution of the code could/should be considered a bug.